### ISSUE TITLE
When client closes pika consumption continues

### ISSUE THREAD

**Issue by skrzepto** [fedora-infra/fmn.sse#1]
When establishing a connection, the data is being sent fine. But once the client disconnect, the server continues to consume messages and send write them.

**Comment by skrzepto** [comment]
http://twistedmatrix.com/documents/12.1.0/web/howto/web-in-60/interrupted.html

looks like something like this may help

`request.notifyFinish().addErrback(self.lc.stop())`

**Comment by skrzepto** [comment]
fixed by 6822110b139e4d3204b954617de20ffa8703a5b3

**Comment by skrzepto** [comment]
For some reason this issue persists

**Comment by skrzepto** [comment]
The issue seems to be this line 

https://github.com/fedora-infra/fmn.sse/blob/develop/fmn/sse/sse_webserver.py#L199

**Comment by skrzepto** [comment]
Figured out the issue. epel 7 had twisted version 16.3.0 while my local f24 copy had 16.2.0

For some reason the functionality breaks in 16.3.0

**Comment by pypingou** [comment]
epel7 has a newer version? :(

**Comment by skrzepto** [comment]
@pypingou I was setting up a fresh env on my vps. When i updated the twisted package in local env it confirmed that the update is causing this issue.

**Comment by skrzepto** [comment]
Opened up a ticket with twisted  https://twistedmatrix.com/trac/ticket/8692

**Comment by skrzepto** [comment]
Temp fix is to stick to 16.2.0 twisted for now until the issue is sorted out

**Comment by skrzepto** [comment]
So twisted labeled the issue as won't fix and don't care about this functionality in new versions :(

**Comment by skrzepto** [comment]
Python Tornado now looks like a good alternative http://www.tornadoweb.org/en/stable/

**Comment by skrzepto** [comment]
After testing the update 16.3.0 again it seems that it still works but its just a lot slower. Is sending out two messages to an inactive connection okay?

**Comment by skrzepto** [comment]
closing since 1-2 messages is acceptable

**Comment by skrzepto** [comment]
https://github.com/fedora-infra/fmn.sse/tree/fix-closing-connections

**Comment by skrzepto** [comment]
I noticed that on an empty queue when a request comes in while no data is
present in the queue and the client closes right away. The connection persists.

It's not until data is added that to the queue that it will close the connection with `twisted>=16.3.0`

**Comment by jeremycline** [comment]
This may or may not still be an issue after the re-write. I've marked it high priority to investigate the problem, and we can re-prioritize or close this issue based on the investigation.

**Comment by jeremycline** [comment]
This should be solved by https://github.com/fedora-infra/fmn.sse/pull/62
